### ISSUE TITLE
too fast update?

### ISSUE THREAD

**Issue by bobybobybob** [TasmoAdmin/TasmoAdmin#1284]
I use TasmoCompiler to make custom Firmware. I download current esp8266 tasmota minimal  (14.4.1) 
I klick on TasmoAdmin on Update, manual update, set the minimal firmware and my new one. 

Modul 13 (Tasmota_Ho)
[2024-12-15T17:20:11.591Z] Starte Update Prozess[2024-12-15T17:20:11.591Z] Checking version...[2024-12-15T17:20:11.809Z] Current version is 14.4.0(TasmoCompiler-esp8266generic)[2024-12-15T17:20:11.809Z] OTA Url => http://192.168.1.18:3004/data/firmwares/tasmota.bin.gz[2024-12-15T17:20:11.920Z] Starte Module update...[2024-12-15T17:20:12.053Z] Sleeping for 10s...[2024-12-15T17:20:22.061Z] Update gestartet[2024-12-15T17:20:22.061Z] Checking version...[2024-12-15T17:20:22.929Z] Version is 14.4.0(TasmoCompiler-esp8266generic)[2024-12-15T17:20:22.929Z] Device upgrade successful

It write, it has "upgrade to 14.4.0, not 14,4,1... but successful....  but that is not true....  when i go to that device it show it is still on minimal firmware. Start there the update again, then it work most time... not everytime... 

No problems with esp32 firmwares  and no problem when automatic is chosen for esp8266. 

sometimes it also helps to select the two own files for update again in tasoadmin but then not to select any devices... the device then fetches the firmware from the local URL again... no idea... in any case with a longer wait it sometimes works. 

I had the same problems a few days ago with 14.4.0 but thought it was due to an outdated minimal. Now I know it wasn't that. 

TasoAdmin 4.2.0 in Docker on Linux Mint in Virtuall Box under windows 10.

**Comment by inverse** [comment]
The whole manual update logic could do with a massive overhaul due to the lack of version checking and assumptions made based on this. 

For the automatic one we use the target version from the OTA URL and use this to compare against the version the device exposes via `http://<ip>/cm?cmnd=Status%200` under `Status.StatusFWR.Version`

For custom frimwares I'm guessing this version is also defined under the endpoint we would just need a way to specify the target version (form input) or extract/infer from the binary some how :thinking:.

**Comment by bobybobybob** [comment]
yes  the problem persists...  even compiled for esp8266 i have to do it again manually via the respective web interface.

**Comment by inverse** [comment]
Does the the device report the correct version on  the API? under `Status.StatusFWR.Version`?

**Comment by bobybobybob** [comment]
I have found out in the meantime that when the program says that the update was successful, the corresponding Tasmota end devices are not yet finished with phase 1 where the minimum is installed first. Simply wait much longer and then phase 2 is also carried out. I don't know why this is the case now, some versions were not like this before.

**Comment by ollo69** [comment]
Same problem here, local update does not works.

**Comment by inverse** [comment]
@ollo69 do you know if the device provides the correct version on the API? e.g. under `Status.StatusFWR.Version`. If it does we could in theory ask for the target version number to be provided or somehow infer it from the binary artifact.

**Comment by ollo69** [comment]
I suppose that it provide correct version because on TasmoAdmin console version column report right version. Was this the question?
